firefox-ios/firefox-ios-tests/Tests/ClientTests/ShareManagerTests.swift

```diff
@@ -212,9 +212,183 @@ final class ShareManagerTests: XCTestCase {
         )
     }
 
+    // MARK: - Sent from Firefox experiment - test that special treatment is only enabled when feature flag is enabled
+
+    func testGetActivityItems_forTab_withSentFromFirefoxEnabled_OverridesURL_withTreatmentA() throws {
+        setupNimbusSentFromFirefoxTesting(isEnabled: true, isTreatmentA: true)
+
+        // TODO: FXIOS-10858 Real links to come
+        let expectedShareContentA = "https://mozilla.org Sent from Firefox 🦊 Try the mobile browser: <FXIOS-10858 marketing link here>"
+        let whatsAppActivityIdentifier = "net.whatsapp.WhatsApp.ShareExtension"
+        let whatsAppActivity = UIActivity.ActivityType(rawValue: whatsAppActivityIdentifier)
+
+        let activityItems = ShareManager.getActivityItems(
+            forShareType: .tab(url: testWebURL, tab: testTab),
+            withExplicitShareMessage: nil
+        )
+
+        // Check we get all 4 types of share items for tabs below
+        XCTAssertEqual(activityItems.count, 4)
```

Comment:
i'd respect here given/when/then paradigm so i'd move all the assertion at the end of the tests.

Reply:
I've made a ticket to update the ShareManager tests here: [FXIOS-10882](https://mozilla-hub.atlassian.net/browse/FXIOS-10882)